### PR TITLE
Custom alphabet for Base64

### PR DESCRIPTION
**Pull request by kkocel**

### CODE REVIEW

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hello @kkocel 
Thank you for your PR.
Please rebase on git master to pick up our JaCoCo configuration.

**Comment by garydgregory** [comment]
And one more rebased on git master please, it looks like we were missing a Checkstyle rule for whitespace. Run `mvn` by itself to run all build checks before pushing. TY!

**Comment by garydgregory** [comment]
@kkocel ping.

**Comment by kkocel** [comment]
Will go back to this during the work week :)

File: src/main/java/org/apache/commons/codec/binary/Base64.java [hunk 1]
```diff
@@ -802,6 +857,23 @@ protected boolean isInAlphabet(final byte octet) {
         return octet >= 0 && octet < decodeTable.length && decodeTable[octet] != -1;
     }
 
+    /**
+     * calculates a decode table for a given encode table
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/codec/binary/Base64.java:861]: Sentences start with a capital letter and end with a period.

File: src/main/java/org/apache/commons/codec/binary/Base64.java [hunk 2]
```diff
@@ -802,6 +857,23 @@ protected boolean isInAlphabet(final byte octet) {
...
+     *
+     * @param encodeTable that is used to determine decode lookup table
+     * @return decodeTable
+     */
+    private byte[] calculateDecodeTable(byte[] encodeTable) {
+        byte[] decodeTable = new byte[256];
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/codec/binary/Base64.java:867]: Refactor magic number to a constant.

File: src/main/java/org/apache/commons/codec/binary/Base64.java [hunk 3]
```diff
@@ -581,13 +596,54 @@ public Base64(final int lineLength, final byte[] lineSeparator, final boolean ur
...
+     *            4). If lineLength &lt;= 0, then the output will not be divided into lines (chunks). Ignored when
+     *            decoding.
+     * @param lineSeparator
+     *            Each line of encoded data will end with this sequence of bytes.
+     * @param encodeTable
+     *            The manual encodeTable - a byte array of 64 chars.
+     * @param decodingPolicy The decoding policy.
+     * @throws IllegalArgumentException
+     *             Thrown when the {@code lineSeparator} contains Base64 characters.
+     * @since 1.15
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/codec/binary/Base64.java:628]: Next version will be 1.17.0.

**Review by garydgregory — CHANGES_REQUESTED** [review]
@kkocel
TY for the PR. Please see my comments.

**Comment by kkocel** [comment]
@garydgregory, when do you plan to release 1.17.0?

**Comment by garydgregory** [comment]
Soonish, maybe within a week.

**Pull request merged**